hail/src/test/scala/is/hail/expr/ir/PruneSuite.scala

```diff
@@ -227,6 +227,15 @@ class PruneSuite extends SparkSuite {
     )
   }
 
+  @Test def testTableMultiWayZipJoinMemo() {
```

Comment:
I'm not sure if this test is necessary or useful. Looking for advice here.

Reply:
@tpoterba ?

Reply:
I think there's a case that's not tested, and will break your prune code --

I do something like

```
(TableMultiWayZipJoin
  TableRead 1
  (TableUnion
    TableRead 2
    (TableFilter
      TableRead 3
    )
  )
)
```

If my requestedType coming into the TMWZJ is not the full type, then I can get into a situation where my requested types passed into my children are identical, but the rebuilt types coming back up are different. This will violate the requirement in the TMWZJ constructor.